### PR TITLE
Reduce header inclusion

### PR DESCRIPTION
**Pull request by janisozaur**
Speed up compilation time by processing less code, improve recompilation cache hit-rate, etc.

See https://gist.github.com/janisozaur/29bd820f55c6c9a9c56a5e20c583f321/revisions for the full list of changes in this PR.

Sorry for the large amount of commits, this can get squashed to just one.

### CODE REVIEW

File: src/openrct2/interface/Viewport.h [hunk 1]
```diff
@@ -17,11 +17,18 @@
 #ifndef _VIEWPORT_H_
 #define _VIEWPORT_H_
 
-#include "../world/Map.h"
-#include "../world/Sprite.h"
-#include "../paint/Paint.h"
 #include "Window.h"
 
+struct paint_session;
```
Comment (AaronVanGeffen) [inline src/openrct2/interface/Viewport.h]: Is this block really worth the reduced inclusions?

**Comment by AaronVanGeffen** [comment]
Nice work. I assume this optimises compilation times a little. (Have you run any benchmarks?)

I do not think it necessary to use one commit per changed file, though. Do you intend for this to be squash-merged?

**Comment by janisozaur** [comment]
> Sorry for the large amount of commits, this can get squashed to just one.

**Comment by janisozaur** [comment]
I have not run any benchmarks, other than verifying it indeed reduce the header inclusion (see the link I attached), overall the diff is on this _graph_ is 73 insertions(+), 666 deletions(-), I count that as enough of a proof of added value.

File: src/openrct2/interface/Viewport.h (hunk 1 above)
Comment (janisozaur) [reply src/openrct2/interface/Viewport.h]: Why would you not want to forward declare things when you can? My answer is: yes.

**Comment by janisozaur** [comment]
With this PR:
make -j 8  155,33s user 17,14s system 724% cpu 23,804 total
make -j 8  154,92s user 17,53s system 732% cpu 23,553 total
make -j 8  154,80s user 17,58s system 724% cpu 23,807 total

Before this PR:
make -j 8  156,22s user 17,64s system 718% cpu 24,183 total
make -j 8  155,98s user 17,56s system 728% cpu 23,810 total
make -j 8  155,88s user 18,17s system 725% cpu 23,995 total

Compilation time is ever so slightly improved, but there's more to be done regarding the included headers. I plan on looking into that later on, but this PR should be considered finished.

File: src/openrct2/interface/Viewport.h (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/interface/Viewport.h]: I agree with @AaronVanGeffen here. One or two is fine, but I think this is too many.

**Comment by IntelOrca** [comment]
@janisozaur what about msvc?

**Comment by janisozaur** [comment]
@IntelOrca I don't have access to it other than AppVeyor. Please test it yourself.

File: src/openrct2/interface/Viewport.h (hunk 1 above)
Comment (IntelOrca) [reply src/openrct2/interface/Viewport.h]: Map, Sprite and Paint are very big headers though, so this is most likely going to have the biggest impact on performance.

Comment (Gymnasiast) [reply src/openrct2/interface/Viewport.h]: Well, all right, I'll leave the decision to you then.

Comment (duncanspumpkin) [reply src/openrct2/interface/Viewport.h]: It's super annoying making changes to anything in the peep or vehicle headers they get included everywhere. Anything that reduces that would be great

File: src/openrct2/core/File.cpp [hunk 2]
```diff
@@ -24,9 +24,9 @@
 #include "File.h"
 #include "FileStream.hpp"
 #include "String.hpp"
+#include "../util/Util.h"
 
 #include "../platform/platform.h"
-#include "../util/Util.h"
```
Comment (Broxzier) [inline src/openrct2/core/File.cpp]: Why this mode?

File: src/openrct2/peep/Peep.h [hunk 3]
```diff
@@ -38,6 +38,8 @@
 #define PEEP_MAX_ENERGY 128
 #define PEEP_MAX_ENERGY_TARGET 255 // Oddly, this differs from max energy!
 
+struct rct_tile_element;
```
Comment (Broxzier) [inline src/openrct2/peep/Peep.h]: I think it'll be nicer to have forward declarations directly after the include list.

File: src/openrct2/core/File.cpp (hunk 2 above)
Comment (janisozaur) [reply src/openrct2/core/File.cpp]: @Broxzier sorry, I did not see your comment until after I merged. Somehow github does not refresh properly today.

That's still a remnant of the `extern "C" {}` times, I think, where the includes had to be split. I must have initially removed this include and manually add it later. Would be solved with https://github.com/OpenRCT2/OpenRCT2/pull/7178

**Pull request merged**